### PR TITLE
Cleanup and conversion to package format 2

### PR DESCRIPTION
**Pull request by ipa-rmb**

### CODE REVIEW

**Comment by floweisshardt** [comment]
FR: OK
@ipa-fxm please do a CR

File: cob_cam3d_throttle/package.xml [hunk 1]
```diff
@@ -12,16 +13,13 @@
   <author email="goa@ipa.fhg.de">Georg Arbeiter</author>
```
Comment (fmessmer) [inline cob_cam3d_throttle/package.xml:13]: change maintainer?

File: cob_cam3d_throttle/package.xml [hunk 2]
```diff
@@ -12,16 +13,13 @@
...
-  <run_depend>pluginlib</run_depend>
-  <run_depend>message_filters</run_depend>
-  <run_depend>nodelet</run_depend>
+  
+  <depend>roscpp</depend>
+  <depend>sensor_msgs</depend>
+  <depend>pluginlib</depend>
+  <depend>message_filters</depend>
+  <depend>nodelet</depend>
+
```
Comment (fmessmer) [inline cob_cam3d_throttle/package.xml:22]: alphabetic order

File: cob_image_flip/package.xml [hunk 3]
```diff
@@ -15,32 +16,19 @@
...
+  <depend>pcl_ros</depend>
+  <depend>cmake_modules</depend>
+  <depend>roscpp</depend>
+  <depend>sensor_msgs</depend>
+  <depend>geometry_msgs</depend>
+  <depend>tf</depend>
+  <depend>image_transport</depend>
+  <depend>cv_bridge</depend>
+  <depend>nodelet</depend>
+  <depend>pcl_conversions</depend>
```
Comment (fmessmer) [inline cob_image_flip/package.xml:29]: alphabetic order

File: cob_object_detection_msgs/package.xml [hunk 4]
```diff
@@ -15,19 +16,14 @@
...
-
-  <run_depend>actionlib_msgs</run_depend>
-  <run_depend>geometry_msgs</run_depend>
-  <run_depend>message_runtime</run_depend>
-  <run_depend>sensor_msgs</run_depend>
-  <run_depend>std_msgs</run_depend>
-  <run_depend>std_srvs</run_depend>
+  <depend>actionlib_msgs</depend>
+  <depend>geometry_msgs</depend>
+  <depend>message_generation</depend>
```
Comment (fmessmer) [inline cob_object_detection_msgs/package.xml:21]: message generation is only build_dependency

File: cob_object_detection_visualizer/package.xml [hunk 5]
```diff
@@ -22,14 +22,11 @@
...
-  <build_depend>cob_object_detection_msgs</build_depend>
-  <build_depend>visualization_msgs</build_depend>
   
-  <run_depend>roscpp</run_depend>
-  <run_depend>cob_object_detection_msgs</run_depend>
-  <run_depend>visualization_msgs</run_depend>
+  <depend>cmake_modules</depend>
+  <depend>roscpp</depend>
+  <depend>cob_object_detection_msgs</depend>
+  <depend>visualization_msgs</depend>
```
Comment (fmessmer) [inline cob_object_detection_visualizer/package.xml:29]: alphabetic order

File: cob_cam3d_throttle/package.xml [hunk 6]
```diff
@@ -12,16 +13,13 @@
...
   <!-- <test_depend>roscpp</test_depend> -->
   <!-- <test_depend>sensor_msgs</test_depend> -->
   <!-- <test_depend>pluginlib</test_depend> -->
```
Comment (fmessmer) [inline cob_cam3d_throttle/package.xml:25]: for cleaning up you can remove the commented lines

File: cob_object_detection_visualizer/package.xml [hunk 7]
```diff
@@ -22,14 +22,11 @@
   <!--   <test_depend>gtest</test_depend> -->
```
Comment (fmessmer) [inline cob_object_detection_visualizer/package.xml:22]: for cleaning up you can remove the commented lines

File: cob_perception_common/package.xml [hunk 8]
```diff
@@ -17,12 +17,12 @@
...
-  <run_depend>cob_object_detection_msgs</run_depend>
-  <run_depend>cob_perception_msgs</run_depend>
-  <run_depend>cob_vision_utils</run_depend>
-  <run_depend>cob_object_detection_visualizer</run_depend>
+  <exec_depend>cob_image_flip</exec_depend>
+  <exec_depend>cob_cam3d_throttle</exec_depend>
+  <exec_depend>cob_object_detection_msgs</exec_depend>
+  <exec_depend>cob_perception_msgs</exec_depend>
+  <exec_depend>cob_vision_utils</exec_depend>
+  <exec_depend>cob_object_detection_visualizer</exec_depend>
```
Comment (fmessmer) [inline cob_perception_common/package.xml:25]: alphabetic order

File: cob_perception_msgs/package.xml [hunk 9]
```diff
@@ -15,15 +16,12 @@
...
-  <build_depend>message_generation</build_depend>
-  <build_depend>sensor_msgs</build_depend>
-  <build_depend>geometry_msgs</build_depend>
-  <build_depend>std_msgs</build_depend>
-
-  <run_depend>message_runtime</run_depend>
-  <run_depend>sensor_msgs</run_depend>
-  <run_depend>geometry_msgs</run_depend>
-  <run_depend>std_msgs</run_depend>
+  <depend>message_generation</depend>
```
Comment (fmessmer) [inline cob_perception_msgs/package.xml:19]: message generation is only build_depend

File: cob_perception_msgs/package.xml [hunk 10]
```diff
@@ -15,15 +16,12 @@
...
+  <depend>sensor_msgs</depend>
+  <depend>geometry_msgs</depend>
+  <depend>std_msgs</depend>
+
```
Comment (fmessmer) [inline cob_perception_msgs/package.xml:23]: alphabetic order

File: cob_vision_utils/package.xml [hunk 11]
```diff
@@ -15,16 +16,11 @@
```
Comment (fmessmer) [inline cob_vision_utils/package.xml:16]: change maintainer

File: cob_vision_utils/package.xml [hunk 12]
```diff
@@ -15,16 +16,11 @@
...
-  <run_depend>visualization_msgs</run_depend>
-  <run_depend>tinyxml</run_depend>
+  <depend>libopencv-dev</depend>
+  <depend>roscpp</depend>
+  <depend>visualization_msgs</depend>
+  <depend>cmake_modules</depend>
+  <depend>tinyxml</depend>
 
   <export>
         <!--cpp cflags="-I${prefix}/common/include -I${prefix}/ros/include" lflags="-Wl,-rpath,${prefix}/common/lib -L${prefix}/common/lib -lcob_vision_utils"/-->
```
Comment (fmessmer) [inline cob_vision_utils/package.xml:26]: remove commented lines

**Comment by fmessmer** [comment]
This PR only affects package.xml files.... did you check whether the CMakeLists.txt files (especially the `find_package` and `catkin_package` tags) have to be adjusted accordingly? see https://github.com/ipa320/care-o-bot/issues/7

**Comment by fmessmer** [comment]
Also, when converting to package-format 2, we agreed to consequently use:
`add_dependencies(TARGET ${${PROJECT_NAME}_EXPORTED_TARGETS})` in case you generate messages within this package and `add_dependencies(TARGET ${catkin_EXPORTED_TARGETS})` in case this package includes **any** package that generates messages/services/actions....

(use `add_dependencies(TARGET ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})` in case both applies!)

Do not use any `*_generate_messages_cpp` or `*_gen_cpp` or alike.

**Pull request merged**